### PR TITLE
Refactor comm create to address the dependency issue during MPID_Init_hook

### PR DESCRIPTION
**Pull request by yfguo**
This PR is a set of commits that addresses the dependency mess in MPID_Init_hook.

## Author Checklist
* [ ] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [ ] Passes tests (included warning check)
* [ ] Confirm whitespace/style checkers are happy (or has a good reason for being bad)
* [ ] Commits are self-contained and do not do two things at once
* [ ] Remove xfail from the test suite when fixing a test
* [ ] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [ ] Add comments such that someone without knowledge of the code could understand
* [ ] Add Devel Docs in the `doc/` directory for any new code design

### CODE REVIEW

File: src/mpid/ch4/netmod/ofi/ofi_pre.h [hunk 1]
```diff
@@ -48,7 +48,7 @@ typedef struct {
 } MPIDI_OFI_comm_t;
 enum {
     MPIDI_AMTYPE_SHORT_HDR = 0,
-    MPIDI_AMTYPE_SHORT,
+    MPIDI_AMYPE_SHORT,
```
Comment (hzhou) [inline src/mpid/ch4/netmod/ofi/ofi_pre.h]: accidental typo?

**Comment by yfguo** [comment]
test:jenkins/ch4/ofi

**Comment by yfguo** [comment]
test:jenkins/ch4/ofi

**Comment by yfguo** [comment]
fixed some lingering problems.
test:jenkins/ch4/ofi

**Comment by yfguo** [comment]
test:jenkins/ch4/ofi

**Comment by yfguo** [comment]
test:jenkins/ch4/ucx

**Comment by yfguo** [comment]
test:jenkins/ch4/ofi
test:jenkins/ch4/ucx

**Pull request closed without merging**